elastic/kibana
# [ResponseOps][flapping] trim collections of recovered alerts persisted for flapping

Resolves https://github.com/elastic/kibana/issues/147205

## Summary

Trimming "recovered" alerts that are stored in the task state to the existing "active" circuit breaker.
Currently we are going to drop these alerts, but once we start working on https://github.com/elastic/kibana/issues/143445 we will need to process these alerts like other flapping alerts that eventually recover and then stop flapping.


### Checklist



### To verify

- You can do Patrick's testing listed in the ticket verify that you see the log statement and that the recovered alerts are being dropped from the task.state 
- You can also modify the maxAlert value here: `x-pack/plugins/alerting/server/alert/create_alert_factory.ts (line 126)` to be something smaller
- Create a rule with a larger number of alerts and verify that you see the warning log statement and that the recovered alerts are being dropped from the task.state
